lib/castle/api/session_sharer.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'singleton'
+
+module Castle
+  module API
+    # this class keeps http session for reuse
```

Comment:
This is misleading, this class only caches the `Net::HTTP` object. We're not doing any persistent TCP/HTTP connection and each request still opens a new TCP connection. 

You can check out how https://github.com/drbrain/net-http-persistent does this

Reply:
yes I actually wanted to expose config object, corrected description and dsl a bit